baybe/utils/dataframe.py

```diff
@@ -360,25 +366,31 @@ def fuzzy_row_match(
 
     # Iterate over all input rows
     for ind, row in right_df.iterrows():
-        # Check if the row represents a valid input
-        valid = True
+        # Check if all values of the row are in the respective parameter ranges
         for param in parameters:
-            if param.is_numeric:
-                if numerical_measurements_must_be_within_tolerance:
-                    valid &= param.is_in_range(row[param.name])
-            else:
-                valid &= param.is_in_range(row[param.name])
-            if not valid:
-                raise ValueError(
-                    f"Input data on row with the index {row.name} has invalid "
-                    f"values in parameter '{param.name}'. "
-                    f"For categorical parameters, values need to exactly match a "
-                    f"valid choice defined in your config. "
-                    f"For numerical parameters, a match is accepted only if "
-                    f"the input value is within the specified tolerance/range. Set "
-                    f"the flag 'numerical_measurements_must_be_within_tolerance' "
-                    f"to 'False' to disable this behavior."
-                )
+            if not param.is_in_range((val := row[param.name])):
+                if param.is_numeric and on_tolerance_violation == "ignore":
+                    break
+                if param.is_numeric and on_tolerance_violation == "warn":
+                    warnings.warn(
+                        f"The value '{val}' is outside the range of parameter "
+                        f"'{param.name}'. "
+                        f"If you expected a match between your input "
+                        f"and the parameter, consider increasing the parameter's "
+                        f"tolerance value or adding more parameter values. "
+                        f"You can silence this warning using the 'ignore' mode.",
+                        UserWarning,
+                    )
+                    break
+                else:
+                    raise ValueError(
+                        f"The value '{val}' is outside the range of parameter "
+                        f"'{param.name}'. "
+                        f"If you expected a match between your input "
+                        f"and the parameter, consider increasing the parameter's "
+                        f"tolerance value or adding more parameter values. "
+                        f"You can bypass this check using the 'ignore' or 'warn' mode."
+                    )
```

Comment:
i cant comment on the line, but we shouldnt forget about whats below here in line 411+

1) it still uses old logger mechanism
2) the warnings there should also be controlled by on_tolerance_validation. eg no need to print a no match found warning if its set to ignore